modules/cms-kit/src/Volo.CmsKit.Admin.Application.Contracts/Volo/CmsKit/Admin/Tags/EntityTagSetDto.cs

```diff
@@ -1,15 +1,41 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Volo.CmsKit.Localization;
+using Volo.CmsKit.Tags;
 
 namespace Volo.CmsKit.Admin.Tags;
 
 [Serializable]
-public class EntityTagSetDto
+public class EntityTagSetDto : IValidatableObject
 {
     public string EntityId { get; set; }
+
     public string EntityType { get; set; }
+
+    [Required]
     public List<string> Tags { get; set; }
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+    {
+        var l = validationContext.GetRequiredService<IStringLocalizer<CmsKitResource>>();
+
+        foreach (var tag in Tags)
+        {
+            if (tag.Length > TagConsts.MaxNameLength)
+            {
+                yield return new ValidationResult(
+                    l[
+                        "MaxTagLengthExceptionMessage",
+                        nameof(tag),
+                        TagConsts.MaxNameLength,
+                        typeof(EntityTagSetDto).FullName
```

Comment:
realy need this full name in Validation message?

Reply:
I think it seems better to find the error easily.

Reply:
I think the end-user do not need this detail info.